worlds/_manual/__init__.py

```diff
@@ -191,6 +223,31 @@ def generate_basic(self):
                 if len(eligible_items) == 0:
                     raise Exception("Could not find a suitable item to place at %s. No items that match categories %s." % (location["name"], ", ".join(location["place_item_category"])))
 
+            if "dont_place_item" in location:
+                if len(location["dont_place_item"]) == 0:
+                    continue
+
+                for item in eligible_items:
+                    if item.name in location["dont_place_item"]:
+                        eligible_items = list(filter((item).__ne__, eligible_items))
```

Comment:
Could do this in a single list comprehension instead of a for + if:
eligible_items = [item for item in eligible_items if item.name not in location["dont_place_item"]]

And can do the same for `dont_place_item_category` below since you have the list of forbidden item names.